controllers/pod-mutator/scheduler/scheduler.go

```diff
@@ -37,59 +34,26 @@ type PodSchedulerSetter struct {
 	SchedulerAnnotationKey string
 
 	client.Client
-	decoder *admission.Decoder
-	log     logr.Logger
+	log logr.Logger
 }
 
-// Check if the Handler interface is implemented.
-var _ admission.Handler = &PodSchedulerSetter{}
-
-// +kubebuilder:webhook:path=/mutate-pods,mutating=true,failurePolicy=ignore,groups="",resources=pods,verbs=create,versions=v1,name=pod-mutator.storageos.com
-
 // NewPodSchedulerSetter returns a new Pod Scheduler mutating admission
 // controller.
-func NewPodSchedulerSetter(k8s client.Client, decoder *admission.Decoder, scheduler string) *PodSchedulerSetter {
+func NewPodSchedulerSetter(k8s client.Client, scheduler string) *PodSchedulerSetter {
 	return &PodSchedulerSetter{
 		SchedulerName:          scheduler,
 		Provisioners:           []string{provisioner.DriverName},
 		SchedulerAnnotationKey: PodSchedulerAnnotationKey,
 		Client:                 k8s,
-		decoder:                decoder,
 		log:                    ctrl.Log,
 	}
 }
 
-// Handle handles an admission request and mutates a pod object in the request.
-func (p *PodSchedulerSetter) Handle(ctx context.Context, req admission.Request) admission.Response {
-	pod := &corev1.Pod{}
-
-	err := p.decoder.Decode(req, pod)
-	if err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-
-	// Get the request namespace. This is needed because sometimes the decoded
-	// pod object lacks namespace info.
-	namespace := req.AdmissionRequest.Namespace
-
-	// Create a copy of the pod to mutate.
-	copy := pod.DeepCopy()
-	if err := p.mutatePodsFn(ctx, copy, namespace); err != nil {
-		p.log.Error(err, "failed to set pod scheduler")
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-
-	marshaledPod, err := json.Marshal(copy)
-	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
-	}
-
-	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledPod)
-}
-
-// mutatePodFn mutates a given pod with a configured scheduler name if the pod
-// is associated with volumes managed by the configured provisioners.
-func (p *PodSchedulerSetter) mutatePodsFn(ctx context.Context, pod *corev1.Pod, namespace string) error {
+// MutatePod mutates a given pod with a configured scheduler name if the pod is
+// associated with volumes managed by the configured provisioners.
+//
+// Errors returned here will block creation of the Pod.
```

Comment:
May not be accurate. It depends on the webhook configuration if the failure is ignored or respected.

Reply:
:+1: Changed wording